pkg/registry/thirdpartyresourcedata/util.go

```diff
@@ -20,9 +20,28 @@ import (
 	"fmt"
 	"strings"
 
+	"k8s.io/kubernetes/pkg/api/unversioned"
 	"k8s.io/kubernetes/pkg/apis/extensions"
 )
 
+func ExtractGroupVersionKind(list *extensions.ThirdPartyResourceList) ([]unversioned.GroupVersion, []unversioned.GroupVersionKind, error) {
+	gvs := []unversioned.GroupVersion{}
+	gvks := []unversioned.GroupVersionKind{}
+	for ix := range list.Items {
+		rsrc := &list.Items[ix]
+		kind, group, err := ExtractApiGroupAndKind(rsrc)
+		if err != nil {
+			return nil, nil, err
+		}
+		for _, version := range rsrc.Versions {
+			gv := unversioned.GroupVersion{Group: group, Version: version.Name}
+			gvs = append(gvs, gv)
```

Comment:
There are no duplicates in this list.  (I can add the de-dup logic as an extra double check if you really want me to.)